File: Stacks and queues/Postfix.java
import java.util.*;

public class Postfix {
    public static int evaluatePostfix(String expr) {
        Stack<Integer> stack = new Stack<>();
        String[] tokens = expr.split(" ");
        
        for (String token : tokens) {
            if (token.matches("\\d+")) { // operand
                stack.push(Integer.parseInt(token));
            } else { // operator
                int b = stack.pop();
                int a = stack.pop();
                switch(token) {
                    case "+": stack.push(a + b); break;
                    case "-": stack.push(a - b); break;
                    case "*": stack.push(a * b); break;
                    case "/": stack.push(a / b); break;
                }
            }
        }
        return stack.pop();
    }

    public static void main(String[] args) {
        String expr = "3 4 + 2 * 7 /"; // ((3+4)*2)/7
        System.out.println(evaluatePostfix(expr)); // Output: 2
    }
}

File: Stacks and queues/Postfix.cpp
#include <bits/stdc++.h>
using namespace std;

int evaluatePostfix(string expr) {
    stack<int> st;
    for (int i = 0; i < expr.length(); i++) {
        char c = expr[i];
        if (c == ' ') continue;
        if (isdigit(c)) {
            st.push(c - '0');
        } else {
            int b = st.top(); st.pop();
            int a = st.top(); st.pop();
            switch(c) {
                case '+': st.push(a + b); break;
                case '-': st.push(a - b); break;
                case '*': st.push(a * b); break;
                case '/': st.push(a / b); break;
            }
        }
    }
    return st.top();
}

int main() {
    string expr = "3 4 + 2 * 7 /"; // ((3+4)*2)/7
    cout << evaluatePostfix(expr) << endl; // Output: 2
}
